lightning/src/routing/router.rs

```diff
@@ -2308,7 +2393,8 @@ where L::Target: Logger {
 
 				'path_walk: loop {
 					let mut features_set = false;
-					if let Some(first_channels) = first_hop_targets.get(&ordered_hops.last().unwrap().0.node_id) {
+					let target = ordered_hops.last().unwrap().0.candidate.target().unwrap_or(maybe_dummy_payee_node_id);
```

Comment:
This is probably the only case that an unwrap is needed. You could do:

```rust
let target = match ordered_hops.last().unwrap().0.candidate.target() {
	Some(target) => target,
	None => break 'path_walk,
};
```

But that means there are now two exits to the loop, where only one has the comment. So maybe best to keep it as it is.